publication-testing/src/main/java/no/unit/nva/publication/StubS3Driver.java

```diff
@@ -30,24 +31,28 @@ public StubS3Driver(String bucketName, List<String> filesInBucket) {
 
     @Override
     public String getFile(String filename) {
-        Stream<String> lines = fileContent(filename);
-        return lines.collect(Collectors.joining(System.lineSeparator()));
+        List<String> lines = fileContent(filename);
+        return String.join(System.lineSeparator(), lines);
     }
 
     public List<String> getAllIonItems() {
         return listFiles(null)
                    .stream()
-                   .flatMap(this::fileContent)
+                   .map(this::fileContent)
+                   .flatMap(Collection::stream)
                    .collect(Collectors.toList());
     }
 
-    private Stream<String> fileContent(String filename) {
-        InputStream inputStream = attempt(() -> IoUtils.inputStreamFromResources(filename))
-                                      .orElseThrow(fail -> fileNotFoundException());
-
-        GZIPInputStream gzipInputStream = attempt(() -> new GZIPInputStream(inputStream)).orElseThrow();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(gzipInputStream, StandardCharsets.UTF_8));
-        return reader.lines();
+    private List<String> fileContent(String filename) {
+        try (InputStream inputStream = attempt(() -> IoUtils.inputStreamFromResources(filename))
+                                           .orElseThrow(fail -> fileNotFoundException());
+            GZIPInputStream gzipInputStream = attempt(() -> new GZIPInputStream(inputStream)).orElseThrow();
+            BufferedReader reader = new BufferedReader(
+                new InputStreamReader(gzipInputStream, StandardCharsets.UTF_8))) {
```

Comment:
put in a try block because PMD is now complaining. Before it was in test and it didn't care